### ISSUE TITLE
Add option to clone repository via SSH

### ISSUE THREAD

**Issue by Garfield96** [ls1intum/Artemis#802]
### Is your feature request related to a problem?
Some people prefer to use SSH instead of HTTPS to clone repositories. Currently, only the HTTPS link can be retrieved within Artemis.

### Describe the solution you'd like
Add the option to switch between HTTPS and SSH link (like in Github).

**Comment by jpbernius** [comment]
Hello @Garfield96,
thanks for raising this feature request. We discussed using SSH for git operations in regards to several aspects of the system. However, we decided to **not implement** SSH based Git interactions for two reasons:

1. Introducing SSH requires us to provide a mechanism for adding / editing SSH Keys, which also need to be kept in sync with our VCS. Password authentication is supported out of the box by the underlying system and is automatically synced with the directory service.
2. Using SSH increases complexity on the student side, as cloning requires generating and configuring SSH beforehand. (Could be optional, but still more settings for inexperienced students.) We see a lot of students starting without prior experience using either git or ssh.

Right now, using SSH for git is a feature for expert users, aware of the underlying VCS system. As this is a very limited number of users, we do not plan to implement and support this feature.

**Workaround:**
However, I can recommend creating a bash script to rewrite the url and clone the repository if this is a feature you want to use.
